packages/frontend/src/js/app/components/map/map.js

```diff
@@ -686,6 +801,202 @@ export default {
+                }
+
+                // Calcul de la position du marqueur
+                // On utilise la méthode pointOnFeature(), qui garantit que le point soit dans le polygone, plutôt que centroid()
+                const { feature } = layer;
+                const markerPosition = pointOnFeature(feature);
+                // Création du marqueur à partir de la long et lat retournées par pointOnFeature()
+                const lon = markerPosition.geometry.coordinates[0];
+                const lat = markerPosition.geometry.coordinates[1];
+                this.circleWithText(
+                    this.map,
+                    [lat, lon],
+                    this.numberOfShantytownsBy.regions[
+                        feature.properties.code
+                    ] || 0,
+                    20,
+                    3,
+                    "region"
+                ).addTo(this.regionalLayer);
+                this.regionalLayer.setStyle({
+                    color: "white",
+                    weight: 8,
+                    opacity: 1
+                });
+            });
+        },
+
+        // Fonction de chargement des données geoJson départementales
+        loadDepartementalData() {
+            this.departementalLayer.getLayers().forEach(layer => {
+                if (layer instanceof L.Marker) {
+                    layer.remove();
+                    return;
+                }
+
+                // Calcul de la position du marqueur
+                // On utilise la méthode pointOnFeature(), qui garantit que le point soit dans le polygone, plutôt que centroid()
+                const { feature } = layer;
+                const markerPosition = pointOnFeature(feature);
+                // Création du marqueur à partir de la long et lat retournées par pointOnFeature()
+                const lon = markerPosition.geometry.coordinates[0];
+                const lat = markerPosition.geometry.coordinates[1];
+
+                const nbSites =
+                    this.numberOfShantytownsBy.departements[
+                        feature.properties.code
+                    ] || 0;
+                const siteLabel = nbSites > 1 ? "sites" : "site";
+                this.circleWithText(
+                    this.map,
+                    [lat, lon],
+                    `<div><strong>${feature.properties.nom}</strong><br/>${nbSites} ${siteLabel}</div>`,
+                    45,
+                    3,
+                    "dept"
+                ).addTo(this.departementalLayer);
+                this.departementalLayer.setStyle({
+                    color: "white",
+                    weight: 8,
+                    opacity: 1
+                });
+            });
+        },
+
+        // Fonction de chargement des marqueurs par commune
+        loadCityData() {
+            // On crée les marqueurs
+            const citiesMarkers = [];
+            for (var key of Object.keys(this.numberOfShantytownsBy.cities)) {
```

Comment:
for of est déprécié, à ma connaissance, il vaut mieux utiliser un `Object.keys().forEach()`

Reply:
Oui, c'est corrigé